src/Config.js

```diff
@@ -63,35 +68,101 @@ export class Config {
     return serverConfiguration;
   }
 
-  static validateOptions({
-    publicServerURL,
-    revokeSessionOnPasswordReset,
-    expireInactiveSessions,
-    sessionLength,
-    defaultLimit,
-    maxLimit,
-    accountLockout,
-    passwordPolicy,
-    masterKeyIps,
-    masterKey,
-    maintenanceKey,
-    maintenanceKeyIps,
-    readOnlyMasterKey,
-    allowHeaders,
-    idempotencyOptions,
-    fileUpload,
-    pages,
-    security,
-    enforcePrivateUsers,
-    schema,
-    requestKeywordDenylist,
-    allowExpiredAuthDataToken,
-    logLevels,
-    rateLimit,
-    databaseOptions,
-    extendSessionOnUse,
-    allowClientClassCreation,
-  }) {
+  /**
+   * Validates the keys that are set in the Parse Server options.
+   *
+   * @param {Array<String>} allowedKeys The keys that are allowed in the Parse Server.
+   * @param {Array<String>} setKeys The keys that are set in the Parse Server options.
+   * @param {String} [keyPath] The path of the key that is being validated.
+   */
+  static validateConfigKeyNames(allowedKeys, setKeys, keyPath) {
+
+    // Internal keys that are not part of the official Parse Server options and should be ignored
+    const internalKeys = [
+      'level',
+      'state',
+      'loggerController',
+      'filesController',
+      'userController',
+      'pushController',
+      'hasPushScheduledSupport',
+      'hasPushSupport',
+      'pushWorker',
+      'pushControllerQueue',
+      'analyticsController',
+      'cacheController',
+      'parseGraphQLController',
+      'liveQueryController',
+      'databaseController',
+      'hooksController',
+      'authDataManager',
+      'schemaCache',
+      'masterKeyIpsStore',
+      'maintenanceKeyIpsStore',
+      'failedConfigKeyVerification',
+      'patternValidator',
+      'applicationId',
+      'database',
+      '_mount',
+      'generateSessionExpiresAt',
+      'generateEmailVerifyTokenExpiresAt',
+      'version',
+      'RateLimitZone',
+      'setIdempotencyFunction',
+
+      // From test files
+      'retryWrites',
+      'customIdSize',
+      'path',
+      'exampleKey',
+      'websocketTimeout',
+      'ops',
+    ];
+
+    for (const key of setKeys) {
+      if (internalKeys.includes(key)) {
+        continue;
+      }
+
+      if (!allowedKeys.includes(key)) {
+        Config.failedConfigKeyVerification = true;
+        console.error(`Error: Unknown Parse Server option '${keyPath ? keyPath + '.' : ''}${key}'.`);
```

Comment:
We should never log to console directly, we have a Logger class for that. Note that I've changed the warnings to errors, because an unknown option will cause an error and prevent server start.